app/com/lynxanalytics/biggraph/graph_api/MetaGraph.scala

```diff
@@ -33,7 +51,7 @@ sealed trait MetaGraphEntity extends Serializable {
     objectStream.writeObject(source.gUID)
     objectStream.writeObject(this.getClass.toString)
     objectStream.close()
-    UUID.nameUUIDFromBytes(buffer.toByteArray)
+    MetaGraphUtil.makeGUID(buffer.toByteArray)
```

Comment:
Maybe "hide" the whole guid computation in a private method. Instead of if, use assert.

Reply:
> Maybe "hide" the whole guid computation in a private method. Instead of if, use assert.

I just moved the `guid` method (field?) to the last place so it remains easy to take in the whole class.